utils/material_plots_2D.py

```diff
@@ -1,80 +1,167 @@
+"""
+This script must be called with python: 'python material_scan_2D.py --{argument} {value}'.
+The output files are saved in data/{outputDir}/name.suffix.
+If no outputDir is specified, it will be data/plots/name.suffix.
+"""
+
 from __future__ import print_function
+
 import argparse
 import math
+import sys
+from os import fspath
+from os.path import expandvars
+from pathlib import Path
 
-import sys,os
-sys.path.append(os.path.expandvars("$FCCSW") + "/Examples/scripts")
-from plotstyle import FCCStyle
 import ROOT
 
+sys.path.append(expandvars("$FCCSW") + "/Examples/scripts")
+from plotstyle import FCCStyle
+
 
 def main():
-    parser = argparse.ArgumentParser(description='Material Plotter')
-    parser.add_argument('--fname', "-f", dest='fname', type=str, help="name of file to read")
-    parser.add_argument('--angleMin', dest='angleMin', default=6, type=float, help="minimum eta/theta/cosTheta")
-    parser.add_argument('--angleMax', dest='angleMax', default=6, type=float, help="maximum eta/theta/cosTheta")
-    parser.add_argument('--angleDef', dest='angleDef', default="eta", type=str, help="angle definition to use: eta, theta or cosTheta, default: eta")
-    parser.add_argument('--angleBinning', "-b", dest='angleBinning', default=0.05, type=float, help="eta/theta/cosTheta bin width")
-    parser.add_argument('--nPhiBins', dest='nPhiBins', default=100, type=int, help="number of bins in phi")
-    parser.add_argument('--x0max', "-x", dest='x0max', default=0.0, type=float, help="Max of x0")                                                                                                                                                
+    parser = argparse.ArgumentParser(description="Material Plotter")
+    parser.add_argument(
+        "--inputFile", "-f", type=str, help="relative path to the input file"
```

Comment:
No pressing need to change the argument from --fname. Changing user interfaces should be avoided unless absolutely necessary., add --inputFile as an alternate if you really want to?

Reply:
I thought of it for two reasons:
1. That way, the argument name is more consistent with [ILDConfig](https://github.com/iLCSoft/ILDConfig/blob/d730298e40b55ff1fa5073a56f9ae510b93be6bc/StandardConfig/production/ILDReconstruction.py#L57). I prefer if scripts name similar arguments similarly
2. The name does not conform to camelCase which most arguments do

If it is used frequently, keeping `--fname` as well is certainly an option

Reply:
Let me say that I can see where you are coming from, but really the big picture matters here, and you have to consider other people as well. Your preferences are not supreme. 

So, no changes to user interfaces unless absolutely necessary. There might be existing documentation somewhere, people have scripts that call this script with this argument. Or the original author just preferred fname and things are as they are, that ship has sailed.

Or as [Linus](https://lkml.org/lkml/2012/12/23/75) puts it "WE DO NOT BREAK USERSPACE!"
(the expletive in that link is not aimed at you), but I have / had to deal with too many callous interface changes in the past.

So, if you want to add `inputName` that is OK, you have a valid argument for that. But you do not have any valid argument to break existing usage.

Reply:
Fair point. Thanks for the explanation. I will make sure that the user interface is not broken :)

Reply:
Thanks for your understanding!